### DISCUSSION TITLE
Frontend (js) network agnostic (average) block time function ect

### DISCUSSION THREAD

**Discussion by temptemp3** [reach-sh/reach-lang#1385] [Ideas]
Hello! I think it would be useful to have a function called something like `blockTime` that returns the average block time. For example, on Algorand this would around 4.3 seconds; however, on Ethereum this would be 10-12 seconds. `blockTime` would return the average number of seconds per block.

**Comment by jeapostrophe** [comment]
I am terrified of implementing this. It is not as simple as hard-coding a number....

https://ycharts.com/indicators/ethereum_average_block_time

https://metrics.algorand.org/#/protocol/#blocks

**Reply by temptemp3** [reply]
Thanks for the lead. All I really need is a provider.

**Reply by temptemp3** [reply]
I am thinking that it wouldn't have to be too complicated. For example,
(1) You could get information about the current block. It has a timestamp.
(2) Then, you could get information about another arbitrary block N blocks prior to block (1).
Then, to compute the average block time, you could divide the difference in timestamp by N.
